File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int buyDay = 0;
        int sellDay = 0;
        int potentialBuy = 0;
        for (int i=1; i<prices.length; ++i) {
            if (prices[i] - prices[potentialBuy] > prices[sellDay] - prices[buyDay]) {
                buyDay = potentialBuy;
                sellDay = i;
            } else if (prices[i] < prices[buyDay]) {
                potentialBuy = i;
            }
        }
        return prices[sellDay] - prices[buyDay];
    }
}

File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        buy = -1
        for i in range(len(prices)-1):
            if buy == -1 and prices[i] < prices[i+1]:
                buy = prices[i]
            elif buy != -1 and prices[i] > prices[i+1]:
                profit += prices[i] - buy
                buy = -1
        if buy != -1:
            profit += prices[len(prices)-1] - buy
        return profit